example/src/main/java/io/netty/example/smtp/SmtpClientHandler.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.netty.example.smtp;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.smtp.DefaultLastSmtpContent;
+import io.netty.handler.codec.smtp.SmtpRequest;
+import io.netty.handler.codec.smtp.SmtpResponse;
+import io.netty.util.concurrent.EventExecutor;
+import io.netty.util.concurrent.FutureCompletionStage;
+import io.netty.util.concurrent.Promise;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * An example smtp client handler
+ */
+public class SmtpClientHandler extends SimpleChannelInboundHandler<SmtpResponse> {
+
+    private ChannelHandlerContext ctx;
+    private final AtomicReference<Promise<SmtpResponse>> responseFuture = new AtomicReference<>();
+
+    @Override
+    public void channelActive(ChannelHandlerContext ctx) throws Exception {
+        this.ctx = ctx;
+    }
+
+    @Override
+    protected void messageReceived(ChannelHandlerContext ctx, SmtpResponse response) throws Exception {
+        Promise<SmtpResponse> promise = responseFuture.get();
+        if (promise == null) {
+            throw new RuntimeException("Unexpected response received: " + response);
+        } else {
+            if (response.code() >= 500) {
+                throw new RuntimeException("receive an error: " + response);
+            }
+            printResponse(response);
+            promise.setSuccess(response);
+            responseFuture.set(null);
+        }
+    }
+
+    private static void printResponse(SmtpResponse response) {
+        System.out.println(response);
+    }
+
+    @Override
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+        cause.printStackTrace();
+        responseFuture.get().setFailure(cause);
+        ctx.close();
+    }
+
+    public FutureCompletionStage<SmtpResponse> send(SmtpRequest request) {
```

Comment:
@yuanrw I think it would be better to just expose `CompletionStage` in general. WDYT ?

Reply:
@normanmaurer Sometimes we may want to block on the result. So I need to get the underlying `Future`.
```
handler.send(req(EHLO, "localhost"))
        .thenCompose(r -> handler.send(req(AUTH, "login")))
        .future().await();
```